### DISCUSSION TITLE
Add option to force IPv4 in httpx backend

### DISCUSSION THREAD

**Discussion by psykokwak-com** [python-telegram-bot/python-telegram-bot#4417] [Feature Suggestions]
It seems that some of us are having issues connecting to the api.telegram.org server through IPv6. Therefore, we want to force the use of IPv4.

I had to change the AsyncHTTPTransport to add a local_address="0.0.0.0" to the constructor in _httpxrequest.py, which allows me to force the use of IPv4 instead of IPv6.

Unfortunately, there is no way to pass this parameter through the PTB library.

**Comment by Bibo-Joshi** [comment]
Hi. Could you kindly elaborate who "we" is?
Please also note that you can implement a custom request class (see [Architecture Overview](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Architecture), [Builder Pattern for Application](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Builder-Pattern), and [telegram.request.BaseRequest](https://docs.python-telegram-bot.org/telegram.request.baserequest.html#telegram.request.BaseRequest)). While this doesn't add an option to the `HTTPXRequest` class, it does allow you to completely customize how your requests are being made

**Comment by psykokwak-com** [comment]
https://github.com/tdlib/telegram-bot-api/issues/556
